### PR TITLE
Code cleanup and formatting for System.Memory src and test

### PR DESCRIPTION
**Pull request by ahsonkhan**
Each set of changes is a separate commit.

Related PR on the coreclr side: https://github.com/dotnet/coreclr/pull/17451

cc @dotnet/corefxlab-contrib, @stephentoub, @jkotas

### CODE REVIEW

File: src/System.Memory/src/System/Buffers/Text/Utf8Parser/Utf8Parser.Date.R.cs [hunk 1]
```diff
@@ -31,13 +31,27 @@ private static bool TryParseDateTimeOffsetR(ReadOnlySpan<byte> source, uint case
...
                 switch (dowString)
                 {
-                    case 0x53756E2c /* 'Sun,' */: dayOfWeek = DayOfWeek.Sunday; break;
-                    case 0x4d6f6e2c /* 'Mon,' */: dayOfWeek = DayOfWeek.Monday; break;
-                    case 0x5475652c /* 'Tue,' */: dayOfWeek = DayOfWeek.Tuesday; break;
-                    case 0x5765642c /* 'Wed,' */: dayOfWeek = DayOfWeek.Wednesday; break;
-                    case 0x5468752c /* 'Thu,' */: dayOfWeek = DayOfWeek.Thursday; break;
-                    case 0x4672692c /* 'Fri,' */: dayOfWeek = DayOfWeek.Friday; break;
-                    case 0x5361742c /* 'Sat,' */: dayOfWeek = DayOfWeek.Saturday; break;
+                    case 0x53756E2c /* 'Sun,' */:
```
Comment (joshfree) [inline src/System.Memory/src/System/Buffers/Text/Utf8Parser/Utf8Parser.Date.R.cs]: I think this change makes the code less readable.

File: src/System.Memory/src/System/Buffers/Text/Utf8Parser/Utf8Parser.Date.R.cs [hunk 2]
```diff
@@ -83,18 +97,42 @@ private static bool TryParseDateTimeOffsetR(ReadOnlySpan<byte> source, uint case
...
-                    case 0x41707220 /* 'Apr ' */ : month = 4; break;
-                    case 0x4d617920 /* 'May ' */ : month = 5; break;
-                    case 0x4a756e20 /* 'Jun ' */ : month = 6; break;
-                    case 0x4a756c20 /* 'Jul ' */ : month = 7; break;
-                    case 0x41756720 /* 'Aug ' */ : month = 8; break;
-                    case 0x53657020 /* 'Sep ' */ : month = 9; break;
-                    case 0x4f637420 /* 'Oct ' */ : month = 10; break;
-                    case 0x4e6f7620 /* 'Nov ' */ : month = 11; break;
-                    case 0x44656320 /* 'Dec ' */ : month = 12; break;
+                    case 0x4a616e20 /* 'Jan ' */ :
```
Comment (joshfree) [inline src/System.Memory/src/System/Buffers/Text/Utf8Parser/Utf8Parser.Date.R.cs]: same for this switch change

**Comment by joshfree** [comment]
```
20:59:42 System\ValueTuple\ValueTuple.cs(249,72): error CS0117: 'HashHelpers' does not contain a definition for 'RandomSeed' [D:\j\workspace\windows-TGrou---2a8f9c29\src\System.ValueTuple\src\System.ValueTuple.csproj]
20:59:42     0 Warning(s)
20:59:42     1 Error(s)

```

While it's great to cleanup code formatting, this type of change is unnecessary risk to the RC milestone.  Please hold off on merging this until master becomes 2.2 later this month.

File: src/Common/src/System/Numerics/Hashing/HashHelpers.cs [hunk 3]
```diff
@@ -6,7 +6,7 @@ namespace System.Numerics.Hashing
 {
     internal static class HashHelpers
     {
-        public static readonly int RandomSeed = Guid.NewGuid().GetHashCode();
+        public static readonly int s_randomSeed = Guid.NewGuid().GetHashCode();
```
Comment (stephentoub) [inline src/Common/src/System/Numerics/Hashing/HashHelpers.cs]: We don't use the s_ naming for public fields.

File: src/System.Memory/src/System/Buffers/Text/Utf8Parser/Utf8Parser.Float.cs [hunk 4]
```diff
@@ -68,7 +68,7 @@ public static bool TryParse(ReadOnlySpan<byte> source, out double value, out int
             if (TryParseNormalAsFloatingPoint(source, out value, out bytesConsumed, standardFormat))
                 return true;
 
-            return TryParseAsSpecialFloatingPoint<double>(source, double.PositiveInfinity, double.NegativeInfinity, double.NaN, out value, out bytesConsumed);
+            return TryParseAsSpecialFloatingPoint(source, double.PositiveInfinity, double.NegativeInfinity, double.NaN, out value, out bytesConsumed);
```
Comment (AtsushiKan) [inline src/System.Memory/src/System/Buffers/Text/Utf8Parser/Utf8Parser.Float.cs]: The float version of this method specifies the generic type explicitly - they should stay consistent. (And I'd prefer you keep it explicit. We don't need to jump to accomodate every time the VS IDE whines that the compiler can figure something out. We don't write source code just for the compiler's sake.)

File: src/System.Memory/src/System/Buffers/Text/Utf8Parser/Utf8Parser.Date.G.cs [hunk 5]
```diff
@@ -2,8 +2,6 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System.Diagnostics;
```
Comment (AtsushiKan) [inline src/System.Memory/src/System/Buffers/Text/Utf8Parser/Utf8Parser.Date.G.cs]: Personally, I don't think we need go out of our way to remove `using System.Diagnostics`. Encouraging the use of assertions by removing friction from its use is more valuable than removing every bit of white-font from the IDE.

File: src/Common/src/System/Numerics/Hashing/HashHelpers.cs (hunk 3 above)
Comment (ahsonkhan) [reply src/Common/src/System/Numerics/Hashing/HashHelpers.cs]: I didn't know that. We should add that as an exception to our [coding style guidelines]( https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md#c-coding-style).

File: src/System.Memory/src/System/Buffers/Text/Utf8Formatter/Utf8Formatter.Date.L.cs [hunk 6]
```diff
@@ -22,7 +22,7 @@ private static bool TryFormatDateTimeL(DateTime value, Span<byte> destination, o
                 return false;
             }
 
-            var dayAbbrev = DayAbbreviationsLowercase[(int)value.DayOfWeek];
+            var dayAbbrev = s_dayAbbreviationsLowercase[(int)value.DayOfWeek];
```
Comment (stephentoub) [inline src/System.Memory/src/System/Buffers/Text/Utf8Formatter/Utf8Formatter.Date.L.cs]: var => whatever type this is... same for the other such cases

**Comment by stephentoub** [comment]
@ahsonkhan, are you still working on this?

**Comment by ahsonkhan** [comment]
> are you still working on this?

I will refresh the PR to address the feedback in a couple of days. Thanks for the reminder.

**Comment by ahsonkhan** [comment]
@dotnet-bot test NETFX x86 Release Build
@dotnet-bot test Windows x64 Debug Build

**Comment by ahsonkhan** [comment]
@dotnet-bot test Windows x64 Debug Build

**Pull request merged**